gax-java/gax/src/test/java/com/google/api/gax/batching/BatcherImplTest.java

```diff
@@ -862,16 +863,28 @@ public void testThrottlingBlocking() throws Exception {
             flowController,
             callContext)) {
       flowController.reserve(1, 1);
+      List<Thread> batcherAddThreadHolder = Collections.synchronizedList(new ArrayList<>());
       Future future =
           executor.submit(
               new Runnable() {
                 @Override
                 public void run() {
+                  batcherAddThreadHolder.add(Thread.currentThread());
                   batcher.add(1);
                 }
               });
-      // Add a little delay ensuring that the next step starts after batcher.add(1)
-      Thread.sleep(10);
+
+      // Wait until batcher.add blocks (Thread.State.WAITING) and the batcher starts the
+      // stopwatch for total_throttled_time. Without this proper waiting, the
+      // Thread.sleep(throttledTime) below may start before the stopwatch starts,
+      // resulting in a shorter total_throttled_time at the verification of throttledTime
+      // at the end of the test.
+      // https://github.com/googleapis/sdk-platform-java/issues/1193
+      do {
+        Thread.sleep(10);
+      } while (batcherAddThreadHolder.isEmpty()
```

Comment:
Should you have some kind of timeout for the waiting? Otherwise, there is a risk of waiting indefinitely.

Reply:
Good point. I just added timeout to this test method.